pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -557,6 +566,60 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
         applyNCMultiQubitOp(arr, num_qubits, matrix, {}, {}, wires, inverse);
     }
 
+    template <class PrecisionT>
+    static void
+    applyPauliRot(std::complex<PrecisionT> *arr, std::size_t num_qubits,
+                  const std::vector<std::size_t> &wires, const bool inverse,
+                  PrecisionT angle, const std::string &word) {
+        using ComplexT = std::complex<PrecisionT>;
+        constexpr std::size_t one{1};
+        constexpr auto IMAG = Pennylane::Util::IMAG<PrecisionT>();
+        PL_ABORT_IF_NOT(wires.size() == word.size(),
+                        "wires and word have incompatible dimensions.")
+        const PrecisionT c = std::cos(angle / 2);
+        const ComplexT s = ((inverse) ? IMAG : -IMAG) * std::sin(angle / 2);
+        const std::array<ComplexT, 4> sines = {s, IMAG * s, -s, -IMAG * s};
+
+        auto get_mask =
+            [num_qubits,
+             &wires]([[maybe_unused]] const std::function<bool(const int)>
+                         &condition) {
+                std::size_t mask{0U};
+                for (std::size_t iw = 0; iw < wires.size(); iw++) {
+                    const auto bit = static_cast<std::size_t>(condition(iw));
+                    mask |= bit << (num_qubits - 1 - wires[iw]);
+                }
+                return mask;
+            };
+        const std::size_t mask_xy =
+            get_mask([&word](const int a) { return word[a] != 'Z'; });
+        const std::size_t mask_y =
+            get_mask([&word](const int a) { return word[a] == 'Y'; });
+        const std::size_t mask_z =
+            get_mask([&word](const int a) { return word[a] == 'Z'; });
+        const auto count_mask_y = std::popcount(mask_y);
+
+        PL_LOOP_PARALLEL(1)
+        for (std::size_t i0 = 0; i0 < (one << num_qubits); i0++) {
```

Comment:
```suggestion
        const std::size_t sv_length = one << num_qubits;
        for (std::size_t i0 = 0; i0 < sv_length; i0++) {
```

Reply:
Keeping, but I'll use `exp2` instead of bit shift.